src/Data/Nat/Properties.agda

```diff
@@ -1728,6 +1736,25 @@ pred-mono-≤ {suc _} {suc _} m≤n = s≤s⁻¹ m≤n
 pred-mono-< : .{{NonZero m}} → m < n → pred m < pred n
 pred-mono-< {m = suc _} {n = suc _} = s<s⁻¹
 
+pred-cancel-≤ : pred m ≤ pred n → (m ≡ 1 × n ≡ 0) ⊎ m ≤ n
+pred-cancel-≤ {m = zero}  {n = zero}  _  = inj₂ z≤n
+pred-cancel-≤ {m = zero}  {n = suc _} _  = inj₂ z≤n
+pred-cancel-≤ {m = suc _} {n = zero} z≤n = inj₁ (refl , refl)
+pred-cancel-≤ {m = suc _} {n = suc _} le = inj₂ (s≤s le)
+
+pred-cancel-< : pred m < pred n → m < n
+pred-cancel-< {m = zero}  {n = suc _} _ = z<s
+pred-cancel-< {m = suc _} {n = suc _}   = s<s
+
+pred-injective : .{{NonZero m}} → .{{NonZero n}} → pred m ≡ pred n → m ≡ n
```

Comment:
No `→` between `NonZero` proofs needed

Reply:
Actually, it's funny. 
If you remove the arrow, Agda throws an error:
```agda
/opt/AGDA/RAMDISK/Agda-stdlib-dev-v2.0-rc1/src/Data/Nat/Properties.agda:1749,19-32
⦃ NonZero m ⦄ cannot appear by itself. It needs to be the argument
to a function expecting an instance argument.
when scope checking ⦃ NonZero m ⦄
```
If, instead, you supply a leading forall quantifier, then you get:
```agda
/opt/AGDA/RAMDISK/Agda-stdlib-dev-v2.0-rc1/src/Data/Nat/Properties.agda:1749,57-58
Variable m is declared irrelevant, so it cannot be used here
when checking that the expression m has type ℕ
```
So the alternative might to introduce the redundant `.{{_ : ...}}` notation, but since the subsequent operations don't depend on such irrelevant proofs, I find it... clutter.

Reply:
Or is this a bug?

Reply:
The leading quantifier error message certainly looks like a bug to me as `m` is not declared irrelevant. You could report it to Agda? However, I was wrong in my initial comment - you do need the arrow :sweat_smile: Sorry!

Reply:
Thanks! ;-)
I'll try to rig up a MWE to post on Agda, but... later.